src/Autofac.Integration.Mef/RegistrationExtensions.cs

```diff
@@ -25,13 +25,13 @@ public static class RegistrationExtensions
     /// Reference to the internal <see cref="Type"/> for <c>System.ComponentModel.Composition.ContractNameServices</c>,
     /// which is responsible for mapping types to MEF contract names.
     /// </summary>
-    private static readonly Type ContractNameServices = typeof(ExportAttribute).Assembly.GetType("System.ComponentModel.Composition.ContractNameServices", true);
+    private static readonly Type ContractNameServices = typeof(ExportAttribute).Assembly.GetType("System.ComponentModel.Composition.ContractNameServices", true)!;
 
     /// <summary>
     /// Reference to the property <c>System.ComponentModel.Composition.ContractNameServices.TypeIdentityCache</c>,
     /// which holds the dictionary of <see cref="Type"/> to <see cref="string"/> contract name mappings.
     /// </summary>
-    private static readonly PropertyInfo TypeIdentityCache = ContractNameServices.GetProperty("TypeIdentityCache", BindingFlags.GetProperty | BindingFlags.Static | BindingFlags.NonPublic);
+    private static readonly PropertyInfo TypeIdentityCache = ContractNameServices.GetProperty("TypeIdentityCache", BindingFlags.GetProperty | BindingFlags.Static | BindingFlags.NonPublic) ?? throw new InvalidOperationException("ContractNameServices.TypeIdentityCache not found.");
```

Comment:
Out of interest, why null-bang the previous Type reference but not this property reference? Just Type vs Property?

Reply:
On the `Type`, the `true` parameter will automatically throw if the type isn't found, so it'll never actually return null.

There isn't an equivalent parameter for `GetProperty` so I fudged in the rough equivalent functionality.

On types that are Autofac.Mef types, I figured using the `nameof` operator would catch any refatoring/rename errors automatically via the compiler, so it wasn't worth adding the extra `throw`.